homeassistant/components/alarmdecoder/alarm_control_panel.py

```diff
@@ -94,8 +100,21 @@ def __init__(self, client, auto_bypass, code_arm_required, alt_night_mode):
         self._ready = None
         self._zone_bypassed = None
         self._auto_bypass = auto_bypass
-        self._code_arm_required = code_arm_required
+        self._attr_code_arm_required = code_arm_required
         self._alt_night_mode = alt_night_mode
+        self._attr_extra_state_attributes = {
+            "ac_power": self._ac_power,
+            "alarm_event_occurred": self._alarm_event_occurred,
+            "backlight_on": self._backlight_on,
+            "battery_low": self._battery_low,
+            "check_zone": self._check_zone,
+            "chime": self._chime,
+            "entry_delay_off": self._entry_delay_off,
+            "programming_mode": self._programming_mode,
+            "ready": self._ready,
+            "zone_bypassed": self._zone_bypassed,
+            "code_arm_required": self._attr_code_arm_required,
+        }
```

Comment:
This is not right, the extra_states_attributes will never be updated, drop this and restore the removed `extra_state_attributes` below.